src/components/Todo.js

```diff
@@ -21,10 +25,17 @@ const Todo = (props) => {
   };
 
   const detailHandler = () => {
-    return <detailForm id={id} details="details" />;
+    console.log("called detail handler",id)
+    if(showDetails === id){
+      setShowDetails(null)
+    }else if(id){
+      setShowDetails(id)
+    }
+    // return <detailForm id={id} details="details" />;
```

Comment:
the returned jsx shall become a block in the component, returning from the event handler will not add it in a dom tree so it will not work. @raiab